### PR TITLE
add PostList, posts store #5

### PR DESCRIPTION
**Pull request by Sota-hub**

### CODE REVIEW

File: helper/descendSort.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const descendSort = (arr) => {
```
Comment (SOunit) [inline helper/descendSort.js:1]: lets add "date"

File: navigation/HomeNavigator.js [hunk 2]
```diff
@@ -1,26 +1,45 @@
...
+      screen: PostList,
+      navigationOptions: ({ navigation }) => {
+        return {
+          headerLeft: leftHeader(navigation),
+        };
+      },
+    },
+    PostDetail: {
+      screen: PostDetail,
+      navigationOptions: ({ navigation }) => {
```
Comment (SOunit) [inline navigation/HomeNavigator.js:34]: maybe stackNavigator offer default go back button without <HeaderBackButton />
check pls?

File: helper/descendSort.js (hunk 1 above)
Comment (Sota-hub) [reply helper/descendSort.js]: I fixed it.

File: screens/PostList.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import descendDateSort from "../helper/descendDateSort";
+import { useDispatch } from "react-redux";
+import { storePosts } from "../store/actions/posts";
+
+const PostList = ({ navigation }) => {
+  const dispatch = useDispatch();
+  const [postsData, setPostsData] = useState();
+
+  useEffect(() => {
+    const db = getDatabase();
```
Comment (SOunit) [inline screens/PostList.js:15]: // firebase setup

File: screens/PostList.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      setPostsData(descendDateSort(data));
+      dispatch(storePosts(data));
+    });
+  }, []);
+
+  return (
+    <ScrollView style={{ width: "100%" }}>
+      {postsData?.map((post, i) => (
+        <TouchableOpacity
+          key={`posts-${i}`}
```
Comment (SOunit) [inline screens/PostList.js:28]: we shouldn't use index as a key

File: store/actions/posts.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export const STOREPOSTS = "get_post";
```
Comment (SOunit) [inline store/actions/posts.js:1]: STORE_POSTS = "STORE_POSTS"

File: store/reducers/posts.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { STOREPOSTS } from "../actions/posts";
+
+const initialState = [];
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case STOREPOSTS:
+      return [action.payload];
```
Comment (SOunit) [inline store/reducers/posts.js:8]: not [...action.payload]?

**Review by SOunit — CHANGES_REQUESTED** [review]
i found some errors
pls fix them

File: screens/PostList.js (hunk 4 above)
Comment (SOunit) [reply screens/PostList.js:28]: for react rendering performance, we should use id for key, not index number in loop

**Comment by Sota-hub** [comment]
I fixed them.

**Review by SOunit — APPROVED** [review]
lgtm

**Comment by SOunit** [comment]
#5 close

**Pull request closed without merging**